resources/main.mjs

```diff
@@ -348,10 +348,24 @@ class MainBenchmarkClient {
     }
 
     _setLocationHash(hash) {
-        if (hash === "#home" || hash === "")
+        if (hash === "#home" || hash === "") {
+            hash = "#home";
             this._removeLocationHash();
-        else
+        } else {
             window.location.hash = hash;
+        }
+        this._updateDocumentTitle(hash);
+    }
+
+    _updateDocumentTitle(hash) {
+        let title = "Speedometer 3";
+        const maybeSection = document.querySelector(hash);
+        if (maybeSection) {
+            const sectionTitle = maybeSection.getAttribute("data-title");
+            if (sectionTitle)
+                title += ` ${sectionTitle}`;
+        }
+        document.title = title;
```

Comment:
Can we instead just have this instead?:
document.title = `Speedometer 3 ${sectionTitle}`

Reply:
Not all sections have titles, and some kind of non-space delimiter is probably nice so I think the conditional is helpful. But if we wanted to golf this function down I think this would work

```
let sectionTitle = document.querySelector(hash)?.getAttribute("data-title") || "";
document.title = `Speedometer 3${sectionTitle && ` - ${sectionTitle}`}`
```

Reply:
used simpler code and `trimEnd`